src/rebar_app_discover.erl

```diff
@@ -16,7 +16,33 @@ do(State, LibDirs) ->
     Apps = find_apps(Dirs, all),
     ProjectDeps = rebar_state:deps_names(State),
     DepsDir = rebar_dir:deps_dir(State),
+    CurrentProfiles = rebar_state:current_profiles(State),
+
+    %% There may be a top level src which is an app and there may not
+    %% Find it here if there is, otherwise define the deps parent as root
+    TopLevelApp = case ec_lists:find(fun(X) ->
+                                             ec_file:real_dir_path(rebar_app_info:dir(X)) =:=
+                                                 ec_file:real_dir_path(rebar_dir:root_dir(State))
+                                     end, Apps) of
+                      {ok, App} ->
+                          rebar_app_info:name(App);
+                      error ->
+                          root
```

Comment:
Eh, we can, not really an issue since app names are binaries, so even if one is named root they won't conflict.